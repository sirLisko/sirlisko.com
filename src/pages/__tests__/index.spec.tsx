import React from "react";
import { render } from "@testing-library/react";
import { StaticQuery } from "gatsby";

import Index from "../index";

jest.mock("../../../data/me.ts", () => ({
  descriptions: ["foo", "bar"],
  keywords: ["k_foo", "k_bar"],
  links: [
    { name: "foo", label: "Foo", url: "https://foo" },
    { name: "bar", label: "Bar", url: "https://Bar" },
  ],
}));

const mockedStaticQuery = StaticQuery as jest.Mock;

beforeEach(() => {
  mockedStaticQuery.mockImplementationOnce(({ render }) =>
    render({
      site: {
        siteMetadata: {
          title: `I am the title`,
        },
      },
    })
  );
});

describe("Index Page", () => {
  it("should render properly", () => {
    const { asFragment } = render(<Index />);
    expect(asFragment()).toMatchSnapshot();
  });
});
