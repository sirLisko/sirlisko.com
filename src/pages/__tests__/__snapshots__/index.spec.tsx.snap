// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index Page should render properly 1`] = `
<DocumentFragment>
  <div
    class="home"
  >
    <div>
      <section
        class="life"
      >
        <span
          class="life__heart"
        />
        <span
          class="life__heart"
        />
        <span
          class="life__heart"
        />
      </section>
      <figure
        class="ghost"
        style="left: -200px; top: -300px;"
      />
    </div>
    <header>
      <ul
        class="links"
      >
        <li
          class="links--foo"
        >
          <a
            href="https://foo"
            title="foo"
          >
            <p
              class="caption"
            >
              Foo
            </p>
          </a>
        </li>
        <li
          class="links--bar"
        >
          <a
            href="https://Bar"
            title="bar"
          >
            <p
              class="caption"
            >
              Bar
            </p>
          </a>
        </li>
      </ul>
    </header>
    <article>
      <section
        class="greeting"
      >
        <img
          alt="sirlisko face"
          class="greeting__face"
          src="./images/sirlisko.svg"
        />
        <div
          class="greeting__intro"
        >
          <p>
            Ciao!
          </p>
          <p>
            I’m sirLisko,
          </p>
          <p>
            another f@$#&n’
          </p>
          <p
            class="greeting__intro__desc"
          >
            <span
              class="greeting__intro--active"
            >
              foo
            </span>
            <span
              class=""
            >
              bar
            </span>
          </p>
        </div>
      </section>
    </article>
    <footer>
      <a
        class="my__projects"
        href="/projects"
      >
        Projects
      </a>
      <a
        class="my__resume"
        href="/resume"
      >
        Resume
      </a>
    </footer>
  </div>
</DocumentFragment>
`;
